modules/routing/routing.cc

```diff
@@ -144,6 +146,41 @@ double Routing::GetRoutingLength(const RoutingResponse& routing_response) {
   return length;
 }
 
+bool Routing::GetParkingID(const PointENU& parking_point,
+                           std::string* parking_space_id) {
+  // search current parking space id associated with parking point.
+  constexpr double kDistance = 0.01;  // meter
+  std::vector<ParkingSpaceInfoConstPtr> parking_spaces;
+  if (hdmap_->GetParkingSpaces(parking_point, kDistance, &parking_spaces) ==
+      0) {
+    *parking_space_id = parking_spaces.front()->id().id();
+    return true;
+  }
+  return false;
+}
+
+bool Routing::FillParkingID(RoutingResponse* routing_response) {
+  // set parking space ID when
+  //  has parking info && has parking point && NOT has parking space id && get
+  //  ID successfully
+  const auto& routing_request = routing_response->routing_request();
+  const bool has_parking_info = routing_request.has_parking_info();
+  if (!has_parking_info ||
+      routing_request.parking_info().has_parking_space_id() ||
```

Comment:
if (has_parking_space_id) , shall return true

Reply:
I think she means no need to fill parking space id anymore if already provided.

Reply:
IMHO, returning true here can indicate that true always means routing response has parking id filled. Might be better from caller's perspective; otherwise, caller needs to check has_parking_space_id again.

Reply:
@unacao Yes. That is what I mean. Thank you for the explanation.
@jmtao @vlin17 In that case, I think the previous "three if's" version is better. if there is no parking info, there is no need to move on check parking id or parking point. So it returns false. And if there is parking id, it returns true. And then check parking_point ... I think it's the best way to organize the logic but I am open for any comments or suggestions anyway.

Reply:
the caller only cares about if "parking_space_id" has been filled up or not.  they don't have to know the logic inside, when and how you process this id. 
true means YES (=parking_space_id is in place fine). 
if you return FALSE, what do you expect the caller to do? unless the caller cares about if the parking_space_id was originally there, or filled by this method indeed.  I don't think that's the case here.